### PR TITLE
Changed description for "branch" and added links

### PR DESCRIPTION
**Pull request by YanisaHS**
### Description

I modified the "branch" concept in the "Git" topic. I thought the basic description could be elaborated on for learners. After that, I went on a short journey of adding links.

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/git/concepts/merge/merge.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ CatalogContent:
   - 'learn-the-command-line'
 ---
 
-In Git, completed changes made to a branch can be merged to the `main` branch. When collaborating on a remote repository, a pull request will need to be opened in order to do this. Anyone with push access to the repository can complete the merge.
+In Git, completed changes made to a [branch](https://www.codecademy.com/resources/docs/git/branch) can be merged to the `main` branch. When collaborating on a remote repository, a pull request will need to be opened in order to do this. Anyone with push access to the repository can complete the merge.
```
Comment (yangc95) [inline content/git/concepts/merge/merge.md:15]:
```suggestion
In Git, completed changes made to a [branch](https://www.codecademy.com/resources/docs/git/branch) can be merged to the `main` branch. When collaborating on a remote repository, a [pull request](https://www.codecademy.com/resources/docs/git/pull-requests) will need to be opened in order to do this. Anyone with push access to the repository can complete the merge.
```

**Review by yangc95 — COMMENTED** [review]
@YanisaHS Conducted a first review! Thanks for these changes ^^

**Comment by YanisaHS** [comment]
@yangc95 Thanks for your feedback! Done! 🥳

**Review by yangc95 — APPROVED** [review]
Looks good to me 👍

**Review by KTom101 — APPROVED** [review]
@YanisaHS 👍

**Pull request merged**